src/instruction_handlers/storage.rs

```diff
@@ -52,6 +61,18 @@ fn sload<T: Tracer>(
 ) -> ExecutionStatus {
     instruction_boilerplate::<opcodes::StorageRead, _>(vm, world, tracer, |vm, args, world| {
         let key = Register1::get(args, &mut vm.state);
+
+        if !vm
+            .world_diff
+            .read_storage_slots_ct
+            .contains(&(vm.state.current_frame.address, key))
+            && !vm
+                .world_diff
+                .written_storage_slots_ct
+                .contains(&(vm.state.current_frame.address, key))
```

Comment:
Same condition twice.

Reply:
no it is not the same it needs to check both read and write storage slots